src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -88,6 +94,96 @@ g.test('mininum_buffer_binding_size')
     }, !success);
   });
 
+g.test('view_dimension')
+  .params(
+    params()
+      .combine(poptions('type', kBindingTypes))
+      .combine(poptions('viewDimension', [undefined, '2d', 'cube'] as const))
+  )
+  .fn(async t => {
+    const { type, viewDimension } = t.params;
+
+    let success = false;
+    if (
+      viewDimension === undefined ||
+      (viewDimension in kTextureViewDimensionInfo && type in kTextureBindingTypeInfo)
```

Comment:
`viewDimension in kTextureViewDimensionInfo` is always true

Reply:
OK, removed this.

> High-level: I think we should try to combine these all into a single test along these lines:
> 
> ```ts
> g.test('entry')
>   .params(
>     params()
>       .combine(poptions('type', kBindingTypes))
>       .combine([
>         ... poptions('visibility', kShaderStageCombinations),
>         ... poptions('hasDynamicOffset', [undefined, false, true]),
>         ... poptions('minBufferBindingSize', [undefined, 0, 4]),
>         ... poptions('textureComponentType', [undefined, ...kTextureComponentTypes]),
>         ... params()
>           .combine(poptions('multisampled', [undefined, false, true]))
>           .combine(poptions('viewDimension', [undefined, ...kTextureViewDimensions])),
>         ... poptions('storageTextureFormat', [undefined, ...kTextureFormats]),
>       ])
>   )
>   .fn(t => {
>     const bglEntry = { /* ... */ };
>     const success = validateBindGroupLayoutEntry(bglEntry);
>     // ...
>   });
> ```
> 
> But this can be a followup.

OK, will do this with a follow-up change, including the validation changes just submitted these days in spec